Makefile

```diff
@@ -307,3 +345,72 @@ verify-bundle: bundle ## Verify bundle update.
 .PHONY: verify-fmt
 verify-fmt: fmt ## Verify fmt update.
 	git diff --exit-code ./api ./controllers
+
+## local configurations
+
+deploy-develmode: manifests kustomize ## Deploy controller in debug mode to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${OPERATOR_IMAGE}
+	@if [ $(NAMESPACE) != '' ];then \
+		echo "Setting Custom Namespace: $(NAMESPACE)"; \
+		cd $(PROJECT_DIR)/config/default && $(KUSTOMIZE) edit set namespace $(NAMESPACE); \
+	fi
+
+	cd $(PROJECT_DIR) && $(KUSTOMIZE) build config/deploy > $(DEPLOYMENT_FILE)
+
+	$(KUSTOMIZE) build config/default | kubectl apply -f -
+
+# clean up
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${DEFAULT_OPERATOR_IMAGE}
+	@if [ $(NAMESPACE) != '' ];then \
+		echo "Removing Custom Namespace: $(NAMESPACE)"; \
+		cd $(PROJECT_DIR)/config/default && $(KUSTOMIZE) edit set namespace $(DEFAULT_REPO); \
+	fi
+
+.PHONY: local-cleanup
+local-cleanup:
+	$(MAKE) kind-delete-cluster
+
+.PHONY: local-env-setup
+local-env-setup:
+	$(MAKE) kind-delete-cluster
+	$(MAKE) kind-create-cluster
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
+
+.PHONY: local-setup
+local-setup: export OPERATOR_IMAGE := authorino-operator:dev
+local-setup:
+	$(MAKE) local-env-setup
+	$(MAKE) docker-build
+	echo "Deploying Authorino control plane"
+	$(KIND) load docker-image ${OPERATOR_IMAGE} --name ${KIND_CLUSTER_NAME}
+	$(MAKE) install
+	$(MAKE) deploy
+
+.PHONY: local-redeploy
+local-redeploy: export OPERATOR_IMAGE := authorino-operator:dev
+local-redeploy:
+	$(MAKE) docker-build
+	echo "Deploying Authorino control plane"
+	$(KIND) load docker-image ${OPERATOR_IMAGE} --name ${KIND_CLUSTER_NAME}
+
+	@if [ $(NAMESPACE) != '' ];then \
+		kubectl rollout restart deployment -n $(NAMESPACE) authorino-operator; \
+		echo "Wait for all deployments to be up"; \
+		kubectl -n $(NAMESPACE) wait --timeout=300s --for=condition=Available deployments --all; \
+		else \
+		kubectl rollout restart deployment -n $(DEFAULT_REPO) authorino-operator; \
+		echo  "Wait for all deployments to be up"; \
+		kubectl -n $(DEFAULT_REPO) wait --timeout=300s --for=condition=Available deployments --all; \
+	fi
+
+## kind configuration
+
+KIND_CLUSTER_NAME ?= authorino-local
+
+.PHONY: kind-create-cluster
+kind-create-cluster: kind ## Create the "authorino-local" kind cluster.
+	$(KIND) create cluster --name $(KIND_CLUSTER_NAME) --config utils/kind-cluster.yaml
+
+.PHONY: kind-delete-cluster
+kind-delete-cluster: kind ## Delete the "authorino-local" kind cluster.
+	$(KIND) delete cluster --name $(KIND_CLUSTER_NAME)
```

Comment:
I'm guessing these local cluster-related targets are all to support testing the operator deployed to a custom namespace. Was that the initial motivation?

To test local changes in the operator, I often find myself running the following:

```sh
kind create cluster
make docker-build OPERATOR_IMAGE=authorino-operator:local
kind load docker-image authorino-operator:local
make manifests && sed -e 's/quay.io\/kuadrant\/authorino-operator:latest/authorino-operator:local/' config/deploy/manifests.yaml | kubectl apply -f -
```

Some other times:

```sh
kind create cluster
kubectl create namespace authorino-operator
make manifests install run
```

Reply:
My initial reason for adding  the local cluster related targets was for similarity in the dev process. Both authorino and kuadrant-operator uses the same naming. The steps you stated are fine but they are not documented so when starting in the project it is unclear how to start with a dev cluster.